src/plone/importexport/browser/templates/importexport.pt

```diff
@@ -139,23 +140,73 @@ i18n:domain="plone">
 
       </subfieldset>
 
-      <!-- Advance Import -->
+      <!-- Advanced Import -->
       <subfieldset tal:define="exist python: True">
 
         <h1 i18n:translate="">Import</h1>
 
           <div>
 
             <form action="imports" enctype="multipart/form-data" method="POST">
+              <h5>Custom Import Settings</h5>
+          <div class="formHelp">
+           Use the settings below to customise the import behaviour
+           </div>
+
+<h5>Primary Key<h5>
+ <select name="import_key" class="pat-select2" data-pat-select2="width:20em">
```

Comment:
why is this hard coded? This should be a list of any indexes probably.

Reply:
The usecase is things like custom fields at the user knows are unique like email address or part_number etc. esp on custom content types. Since it has to use queries it makes sense to look at indexes as long as it matches the field definitions from dexterity.

Reply:
It's no longer hard-coded.